packages/web/app/src/lib/hooks/use-local-storage-json.ts

```diff
@@ -0,0 +1,23 @@
+import { useCallback, useState } from 'react';
+
+export function useLocalStorageJson<T>(key: string, defaultValue: T) {
+  const [value, setValue] = useState<T>(() => {
+    const json = localStorage.getItem(key);
+    try {
+      const result = json ? JSON.parse(json) : defaultValue;
```

Comment:
We can also allow providing a zod model here, what do you think?

Reply:
That would be great. I have a git stash with that work started as a matter of fact. I would be happy to finish it. While working I realized that `useLocalStorageJson` is outside the path of the error and so took a technically simpler solution.

How about I put up a refactor PR for `useLocalStorageJson` after we merge this one?

Reply:
sounds good!